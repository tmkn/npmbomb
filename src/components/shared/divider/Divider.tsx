/** @jsx jsx */
import { jsx, css } from "@emotion/react";

import { mq } from "../../../css";

const dividerStyle = css({
    [mq[0]]: {
        height: "0px",
        borderBottom: "1px solid #ececec",
        display: "flex",
        flex: 1
    }
});

interface IDividerProps {
    margin: string;
}

export const Divider: React.FC<IDividerProps> = ({ margin }) => (
    <div css={dividerStyle} style={{ margin }}></div>
);
