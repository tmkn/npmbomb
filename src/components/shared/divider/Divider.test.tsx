import React from "react";
import { render, fireEvent } from "@testing-library/react";

import { Divider } from "./Divider";

describe("<Divider />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(<Divider margin={"1rem"} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
