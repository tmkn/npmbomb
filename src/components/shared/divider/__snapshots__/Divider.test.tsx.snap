// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-khmwet"
    style="margin: 1rem;"
  />
</DocumentFragment>
`;
